### ISSUE TITLE
No error for unreachable code if other clauses in 'switch' don't return

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#8871]
```ts
function f(x) {
    switch (x) {
        default:
            return false;
        case 2:
            200;
    }
	return true;
}
```

I expected an error on reachability on `case 2`, its statement, or `return true`, but we don't correctly check this.

**Comment by DanielRosenwasser** [comment]
The actual semantics indicate that this issue is wrong :smile:
